src/encrypted_sled/record.rs

```diff
@@ -4,26 +4,39 @@ use chacha20poly1305::XNonce;
 use serde::{Deserialize, Serialize};
 use sled::IVec;
 
+use tofn::sdk::api::{deserialize, serialize};
+
+use super::result::{
+    EncryptedDbError::{Deserialization, Serialization},
+    EncryptedDbResult,
+};
+
 /// The value of [super::Db].
 #[derive(Serialize, Deserialize, Debug)]
 pub(super) struct EncryptedRecord {
     encrypted_value: Vec<u8>,
     nonce: [u8; 24],
 }
+
 impl EncryptedRecord {
     pub(super) fn new(encrypted_value: Vec<u8>, nonce: XNonce) -> Self {
         EncryptedRecord {
             encrypted_value,
             nonce: nonce.into(),
         }
     }
+
     /// Convert a [EncryptedRecord] to bytes using serde.
-    pub(super) fn as_bytes(&self) -> bincode::Result<Vec<u8>> {
-        bincode::serialize(&self)
+    pub(super) fn to_bytes(&self) -> EncryptedDbResult<Vec<u8>> {
+        serialize(&self)
+            .map_err(|_| Serialization("Failed to serialize encrypted record".to_string()))
```

Comment:
Unfortunately returning string error messages is too verbose 😞 

Nip: I see that the `Serialization` error message is unique static. In this case you could define `Serialization` as
```rs
#[error("Failed to serialize encrypted record")]
Serialization,
```
and simply call: 
```rs
serialize(&self).map_err(|_| Serialization)
```

This would have to change that if we ever need to return different type of messages in `Serialization` errors, but I prefer to avoid verbosity as long as we can we can.

Reply:
Yeah, I was debating between the two approaches. I'll go with your suggestion. If needed, we can just add another error type, instead of a String argument.